app/src/main/java/com/rkm/tasky/util/storage/preferences/abstraction/UserPreferencesRepository.kt

```diff
@@ -0,0 +1,5 @@
+package com.rkm.tasky.util.storage.preferences.abstraction
+
+interface UserPreferencesRepository {
+    suspend fun saveValue(key: String, value: String)
```

Comment:
Taking a look at what this class is saving in its implemenation (`USER_KEY_ACCESS_TOKEN` and `USER_KEY_RESPONSE_TOKEN`), don’t you think this is more **session information** instead of a user’s preferences? 

And by using a signature where you require a key and a value, need are exposing some implementation details. Now the caller needs access to the keys in order to save some values. Could it not make more sense to pass in a model with all the information and let the implementation of the class handle how to store it?

```
data class AuthInfo(
    val accessToken: String,
    val refreshToken: String,
    val userId: String,
    val fullName: String,
    val email: String
)
```

```
interface SessionStorage {
    suspend fun getSession(): AuthInfo?
    suspend fun setSession(info: AuthInfo?)
}
```